### PR TITLE
Gekidou - Sidebar - Adds actions, queries, db models and rest client

### PR DESCRIPTION
**Pull request by shazm**
#### Summary

Adds local/remote actions, rest client, queries and a model fix.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-38765

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: 

#### Release Note

```release-note

```

### CODE REVIEW

**Review by enahum — APPROVED** [review]

File: app/actions/remote/team.ts [hunk 1]
```diff
@@ -293,6 +293,7 @@ export const handleTeamChange = async (serverUrl: string, teamId: string) => {
     }
 
     const {channels, memberships, error} = await fetchMyChannelsForTeam(serverUrl, teamId);
```
Comment (enahum) [inline app/actions/remote/team.ts:295]: actually this needs to exclude the direct channels

**Review by enahum — CHANGES_REQUESTED** [review]

File: app/actions/remote/team.ts [hunk 2]
```diff
@@ -292,7 +292,12 @@ export const handleTeamChange = async (serverUrl: string, teamId: string) => {
         await operator.batchRecords(models);
     }
 
+<<<<<<< Updated upstream
```
Comment (enahum) [inline app/actions/remote/team.ts:295]: @shazm bad merge?

**Review by enahum — CHANGES_REQUESTED** [review]

File: app/actions/remote/channel.ts [hunk 3]
```diff
@@ -194,16 +196,19 @@ export const fetchMyChannelsForTeam = async (serverUrl: string, teamId: string,
     }
 
     try {
-        let [channels, memberships]: [Channel[], ChannelMembership[]] = await Promise.all([
+        // eslint-disable-next-line prefer-const
+        let [channels, memberships, categoriesWithOrder]: [Channel[], ChannelMembership[], CategoriesWithOrder] = await Promise.all([
```
Comment (larkox) [inline app/actions/remote/channel.ts:200]: Why not use const here?

Comment (larkox) [inline app/actions/remote/channel.ts:200]: Is it possible to remove the type definition? (client functions should be typed already)

**Review by larkox — APPROVED** [review]
LGTM, just two small comments.

**Review by enahum — APPROVED** [review]

**Pull request merged**